### PR TITLE
[10.0][FIX] l10n_it_intrastat_statement: la rettifica può essere legata a un periodo di tipo diverso

### PR DESCRIPTION
**Pull request by SimoRubi**
https://github.com/OCA/l10n-italy/issues/2293 per `10.0`

--
Confermo di aver firmato il CLA https://odoo-community.org/page/cla e di aver letto le linee guida su https://odoo-community.org/page/contributing

### CODE REVIEW

File: l10n_it_intrastat_statement/tests/test_intrastat_statement.py [hunk 1]
```diff
@@ -163,6 +163,30 @@ def test_statement_sale_quarter(self):
         # Last line is section line, for quarter report it should be 64 chars
         self.assertEqual(len(file_content.splitlines()[-1]), 64)
 
+    def test_statement_sale_quarter_adjustment_period(self):
+        """Check that sale adjustments can be in a different period type"""
+        invoice = self._get_intrastat_computed_invoice()
+        month = fields.Date.from_string(invoice.date_invoice).month
+        quarter = 1 + (month - 1) // 3
```
Comment (sergiocorato) [inline l10n_it_intrastat_statement/tests/test_intrastat_statement.py:170]: Operator `//` in python2 exists?

File: l10n_it_intrastat_statement/tests/test_intrastat_statement.py [hunk 2]
```diff
@@ -163,6 +163,30 @@ def test_statement_sale_quarter(self):
...
+        })
+        statement.compute_statement()
+
+        adjustment_vat_code = 'adjust_vat'
+        statement.sale_section2_ids = [(0, 0, {
+            'partner_id': invoice.partner_id.id,
+            'vat_code': adjustment_vat_code,
+            'sign_variation': '+',
+            'year_id': fields.Date.from_string(invoice.date_invoice).year,
+            'month': 1,
```
Comment (sergiocorato) [inline l10n_it_intrastat_statement/tests/test_intrastat_statement.py:183]: `month` perhaps?

File: l10n_it_intrastat_statement/tests/test_intrastat_statement.py (hunk 1 above)
Comment (OpenCode) [reply l10n_it_intrastat_statement/tests/test_intrastat_statement.py:170]: Yes: https://docs.python.org/2.7/reference/lexical_analysis.html#operators

File: l10n_it_intrastat_statement/tests/test_intrastat_statement.py (hunk 2 above)
Comment (SimoRubi) [reply l10n_it_intrastat_statement/tests/test_intrastat_statement.py:183]: Here I am testing that the adjustment can be in a different period type than the `statement` so it doesn't have to be related to the `statement`'s month.
`month` is only used to compute the `quarter` of the `statement`.

File: l10n_it_intrastat_statement/tests/test_intrastat_statement.py (hunk 1 above)
Comment (sergiocorato) [reply l10n_it_intrastat_statement/tests/test_intrastat_statement.py:170]: Thanks

File: l10n_it_intrastat_statement/tests/test_intrastat_statement.py (hunk 2 above)
Comment (sergiocorato) [reply l10n_it_intrastat_statement/tests/test_intrastat_statement.py:183]: In this way (putting always 1 as `month`), in the first three month of the year month would be in the same quarter of statement, is it wanted?

Comment (SimoRubi) [reply l10n_it_intrastat_statement/tests/test_intrastat_statement.py:183]: The code now allows the period stated in the adjustment to be independent from the period of the `statement` and this is what this test case is checking.
Anyway, I understand that this wouldn't be a realistic use-case when they are overlapping, so I changed the adjustment's year; can you check?

BTW, thanks for the review :)

Comment (sergiocorato) [reply l10n_it_intrastat_statement/tests/test_intrastat_statement.py:183]: Ok (I don't use this version anyway)

**Review by sergiocorato — APPROVED** [review]

**Comment by SimoRubi** [comment]
> What a great day to merge this nice PR. Let's do it!
> Prepared branch [10.0-ocabot-merge-pr-2294-by-sergiocorato-bump-patch](https://github.com/OCA/l10n-italy/commits/10.0-ocabot-merge-pr-2294-by-sergiocorato-bump-patch), awaiting test results.

Credo si sia un po' piantato, @sergiocorato posso fare qualcosa?

**Comment by sergiocorato** [comment]
vero, vediamo se serve a qualcosa:
/ocabot merge patch

**Pull request merged**